src/plots/cartesian/set_convert.js

```diff
@@ -598,14 +606,14 @@ module.exports = function setConvert(ax, fullLayout) {
                         ax._categories = categories;
                         ax._categoriesMap = categoriesMap;
                     } else {
-                        ax._categories = [];
-                        ax._categoriesMap = {};
+                        init();
                     }
                 }
+
+                if(!found) init();
```

Comment:
I am wondering about the position of this line in relation to the `var found = false;` and wanted to double check something:
With this setup if the item is not found in the first iteration (when i === 0), the `init` function would be called. If that is correct, then `var found = false;` could be moved inside the `i` loop?

Reply:
You're right! Thanks for spotting that. 

Done in https://github.com/plotly/plotly.js/pull/3565/commits/598a67bce659e737ecb4dfe11a645135b60f31a5 + I added a `break` as one `ax` can only be in one match group at a time.